### DISCUSSION TITLE
How to get agent generated traceId and span Id, programatically

### DISCUSSION THREAD

**Discussion by annaik-mdsol** [open-telemetry/opentelemetry-java-instrumentation#12228] [Q&A] [answered]
Hi I want to get agent generated traceId and spanId programatically. I want to add traceId and span Id in response header.

**Comment by jaydeluca** [comment, marked as answer]
> I want to get agent generated traceId and spanId programatically

You can use the [java SDK API](https://mvnrepository.com/artifact/io.opentelemetry/opentelemetry-api) in your application to get this information.

Example: 

```java
Span span = Span.current();
SpanContext spanContext = span.getSpanContext();

String traceId = spanContext.getTraceId();
String spanId = spanContext.getSpanId();
```
